packages/router/history.ts

```diff
@@ -612,24 +612,12 @@ function getUrlBasedHistory(
   }
 
   function handlePop() {
-    let nextAction = Action.Pop;
+    action = Action.Pop;
     let nextIndex = getIndex();
-
-    if (nextIndex != null) {
-      let delta = nextIndex - index;
-      action = nextAction;
-      index = nextIndex;
-      if (listener) {
-        listener({ action, location: history.location, delta });
-      }
-    } else {
-      warning(
-        false,
-        `You are trying to perform a POP navigation to a location that was not ` +
-          `created by @remix-run/router. This will fail silently in production.  ` +
-          `You should navigate via the router to avoid this situation (instead of ` +
-          `using window.history.pushState/window.location.hash).`
-      );
```

Comment:
This should only apply when blockers were present, so lift it up to the router.  We'll send `delta === null` when we can't determine it